src/java.desktop/share/classes/javax/swing/border/BevelBorder.java

```diff
@@ -90,17 +94,25 @@ public BevelBorder(int bevelType) {
      * will be derived from specified highlight color and
      * bevel outer shadow color and bevel inner shadow color
      * will be derived from specified shadow color.
+     * If bevelType is not of known type that is neither
+     * {@link BevelBorder#RAISED} or {@link BevelBorder#LOWERED},
+     * no bevel border will be painted.
+     *
      * @param bevelType the type of bevel for the border
      * @param highlight the color to use for the bevel highlight
      * @param shadow the color to use for the bevel shadow
      */
     public BevelBorder(int bevelType, Color highlight, Color shadow) {
-        this(bevelType, highlight.brighter(), highlight, shadow, shadow.brighter());
+        this(bevelType, (highlight != null) ? highlight.brighter() : null,
+             highlight, shadow, (shadow != null) ? shadow.brighter() : null);
```

Comment:
The case where the code skips all these parameters and uses the component colors is not documented in this constructor and in the factory methods?